### PR TITLE
Update for GA

### PR DESCRIPTION
**Pull request by bderveni**
- Update before GA
- Add property requirements for posts and puts

@osimarkrogers @mckenzielee

### CODE REVIEW

File: Documentation/Identity/Identity_ClientCredentialClient.md [hunk 1]
```diff
@@ -8,14 +8,14 @@ APIs for creating, getting, updating, and deleting Client Credential Clients
 
 ## Properties
 
-For HTTP requests and responses, the ClientCredentialClientDto object has the following properties and JSON-serialized body: 
+For HTTP requests and responses, the ClientCredentialClient object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
- --- | --- | ---
+ --- | --- | --- | ---
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_ClientCredentialClient.md]: You don't need the added "| ---". It messes up the table.

File: Documentation/Identity/Identity_ClientCredentialClient.md [hunk 2]
```diff
@@ -338,7 +362,7 @@ Id of tenant
 [FromQuery]
 [Optional]
 [Default = ""]
-string[] tags
+string[] tag
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_ClientCredentialClient.md]: Is this an array of strings that is named "tag"?

File: Documentation/Identity/Identity_HybridClient.md [hunk 3]
```diff
@@ -8,19 +8,19 @@ APIs for creating, getting, updating, and deleting Hybrid Clients
...
 ## Properties
 
-For HTTP requests and responses, the HybridClientDto object has the following properties and JSON-serialized body: 
+For HTTP requests and responses, the HybridClient object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
- --- | --- | ---
-AllowOfflineAccess | optional: bool | Specifies whether this client can request refresh tokens, by providing the *offline_access* scopes.
-AllowAccessTokensViaBrowser | optional: bool | Specifies whether this HybridClientDto is allowed to receive access tokens via the browser. This is useful to harden flows that allow multiple response types (e.g. by disallowing a hybrid flow client that is supposed to use code *id_token* to add the *token* response type, thus leaking the token to the browser).
+ --- | --- | --- | ---
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_HybridClient.md]: Don't need the extra "| ---"

File: Documentation/Identity/Identity_IdentityProvider.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ APIs for getting a list of all supported Identity Providers
 For HTTP requests and responses, the IdentityProvider object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
- --- | --- | ---
+ --- | --- | --- | ---
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_IdentityProvider.md]: Don't need the extra "| ---".

File: Documentation/Identity/Identity_ImplicitClient.md [hunk 5]
```diff
@@ -8,18 +8,18 @@ APIs for creating, getting, updating, and deleting Implicit Clients
 
 ## Properties
 
-For HTTP requests and responses, the ImplicitClientDto object has the following properties and JSON-serialized body: 
+For HTTP requests and responses, the ImplicitClient object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
- --- | --- | ---
+ --- | --- | --- | ---
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_ImplicitClient.md]: Don't need the extra "| ---"

File: Documentation/Identity/Identity_Secret.md [hunk 6]
```diff
@@ -426,13 +245,14 @@ Created
...
- `ClientSecretResponseDto`
+ `ClientSecretResponse`
 
 ```json
 {
-  "ClientSecret": "ClientSecret",
-  "SecretId": "SecretId",
-  "Expiration": "2019-04-30T11:35:12.3311769-07:00",
+  "Secret": "Secret",
+  "Id": 0,
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_Secret.md]: Is 0 an example of a secret Id? I guess it changed to an int32...

File: Documentation/Identity/Identity_User.md [hunk 7]
```diff
@@ -8,10 +8,10 @@ APIs for creating, getting, updating, and deleting Users
 
 ## Properties
 
-For HTTP requests and responses, the UserDto object has the following properties and JSON-serialized body: 
+For HTTP requests and responses, the User object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
- --- | --- | ---
+ --- | --- | --- | ---
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_User.md]: Don't need the extra "| ---"

File: Documentation/Identity/Identity_Client.md [hunk 8]
```diff
@@ -70,7 +70,7 @@ Success
 
 ##### Type:
 
- `List[ClientCredentialClientDto]`
+ `List[ClientCredentialClient]`
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_Client.md]: Other places have only List. I tagged those in other comments. I am not sure if there is supposed to be consistent. 
Also, there is an object call ClientCredentialClient?

File: Documentation/Identity/Identity_ClientCredentialClient.md [hunk 9]
```diff
@@ -385,7 +409,7 @@ Success
 
 ##### Type:
 
- `List[ClientCredentialClientDto]`
+ `List`
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_ClientCredentialClient.md]: This is inconsistent with the List[ClientCredentialClient] in Documentation/Identity/Identity_Client.md.

File: Documentation/Identity/Identity_IdentityProvider.md [hunk 10]
```diff
@@ -36,7 +36,7 @@ Returns an IdentityProvider object
 
 ### Request
 
-`GET api/v1-preview/IdentityProviders/{identityProviderId}`
+`GET api/v1/IdentityProviders/{identityProviderId}`
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_IdentityProvider.md]: Does this return all identity providers supported by OCS?

File: Documentation/Identity/Identity_Invitation.md [hunk 11]
```diff
@@ -8,27 +8,27 @@ APIs for creating, getting, updating, and deleting an Invitation
 
 ## Properties
 
-For HTTP requests and responses, the InvitationDto object has the following properties and JSON-serialized body: 
+For HTTP requests and responses, the Invitation object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
- --- | --- | ---
+ --- | --- | --- | ---
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_Invitation.md]: Don't need the extra "| ---"

File: Documentation/Identity/Identity_User.md [hunk 12]
```diff
@@ -923,7 +563,7 @@ Success
 
 ##### Type:
 
- `Newtonsoft.Json.Linq.JObject`
+ `JObject`
 
 ```json
 {}
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_User.md]: I don't think this json formatted {} adds any value. Maybe it is best to describe or leave out?

File: Documentation/Identity/Identity_Invitation.md [hunk 13]
```diff
@@ -8,27 +8,27 @@ APIs for creating, getting, updating, and deleting an Invitation
 
 ## Properties
 
-For HTTP requests and responses, the InvitationDto object has the following properties and JSON-serialized body: 
+For HTTP requests and responses, the Invitation object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
```
Comment (mckenzielee) [inline Documentation/Identity/Identity_Invitation.md]: Probably don't need the 's' on Descriptions.

File: Documentation/Identity/Identity_ClientCredentialClient.md (hunk 2 above)
Comment (bderveni) [reply Documentation/Identity/Identity_ClientCredentialClient.md]: If you pass multiple query parameters named tag, ASP NET Core will parse them into an array. For instance:

/clients?tag=tag1&tag=tag2&tag=tag3

Will be parsed and passed to our controller as ["tag1", "tag2", "tag3"]

We had a discussion about this in the team and decided that it is more intuitive to have the query param name as singular when passing in lists.

File: Documentation/Identity/Identity_User.md (hunk 12 above)
Comment (bderveni) [reply Documentation/Identity/Identity_User.md]: Was just trying to be consistent with this one.

File: Documentation/Identity/Identity_Invitation.md (hunk 13 above)
Comment (bderveni) [reply Documentation/Identity/Identity_Invitation.md]: I most certainly do not.

File: Documentation/Identity/Identity_AzureActiveDirectoryTenant.md [hunk 14]
```diff
@@ -11,7 +11,7 @@ APIs for creating, getting, updating, and deleting Azure Active Directory Tenant
 For HTTP requests and responses, the AzureActiveDirectoryTenant object has the following properties and JSON-serialized body: 
 
 Property | Type | Descriptions
- --- | --- | ---
+ --- | --- | --- | ---
```
[the remark this answers is not in the document]
Comment (bderveni) [reply Documentation/Identity/Identity_AzureActiveDirectoryTenant.md]: Fixed

File: Documentation/Identity/Identity_ClientCredentialClient.md (hunk 1 above)
Comment (bderveni) [reply Documentation/Identity/Identity_ClientCredentialClient.md]: Fixed

File: Documentation/Identity/Identity_HybridClient.md (hunk 3 above)
Comment (bderveni) [reply Documentation/Identity/Identity_HybridClient.md]: Fixed

File: Documentation/Identity/Identity_IdentityProvider.md (hunk 4 above)
Comment (bderveni) [reply Documentation/Identity/Identity_IdentityProvider.md]: Fixed

File: Documentation/Identity/Identity_Invitation.md (hunk 11 above)
Comment (bderveni) [reply Documentation/Identity/Identity_Invitation.md]: Fixed

File: Documentation/Identity/Identity_Invitation.md (hunk 13 above)
Comment (bderveni) [reply Documentation/Identity/Identity_Invitation.md]: Fixed everywhere! Thanks a ton for pointing these out.

File: Documentation/Identity/Identity_ClientCredentialClient.md (hunk 9 above)
Comment (bderveni) [reply Documentation/Identity/Identity_ClientCredentialClient.md]: Fixed these as well

File: Documentation/Identity/Identity_ImplicitClient.md (hunk 5 above)
Comment (bderveni) [reply Documentation/Identity/Identity_ImplicitClient.md]: Resolved

File: Documentation/Identity/Identity_Secret.md (hunk 6 above)
Comment (bderveni) [reply Documentation/Identity/Identity_Secret.md]: Yep. Secret Ids are ints and not guids.

File: Documentation/Identity/Identity_User.md (hunk 7 above)
Comment (bderveni) [reply Documentation/Identity/Identity_User.md]: Resolved

File: Documentation/Identity/Identity_Client.md (hunk 8 above)
Comment (bderveni) [reply Documentation/Identity/Identity_Client.md]: Fixed

**Pull request merged**